OmniGibson/omnigibson/metrics/task_metric.py

```diff
@@ -19,16 +19,44 @@ def start_callback(self, env):
         self.timesteps = 0
         self.render_timestep = og.sim.get_rendering_dt()
 
+        # Store the initial state (true/false) of each predicate for each option
+        self.initial_predicate_states = []
+        for option in env.task.ground_goal_state_options:
+            option_states = []
+            for predicate in option:
+                option_states.append(predicate.evaluate())
+            self.initial_predicate_states.append(option_states)
+
     def step_callback(self, env):
         self.timesteps += 1
 
     def end_callback(self, env):
         candidate_q_score = []
-        for option in env.task.ground_goal_state_options:
-            predicate_truth_values = []
-            for predicate in option:
-                predicate_truth_values.append(predicate.evaluate())
-            candidate_q_score.append(np.mean(predicate_truth_values))
+        for i, option in enumerate(env.task.ground_goal_state_options):
+            # Compare current predicate states with initial states
+            initial_states = self.initial_predicate_states[i]
+
+            # Count predicates that weren't initially satisfied (these are the ones we need to complete)
+            predicates_to_complete = sum(1 for was_satisfied in initial_states if not was_satisfied)
+
+            if predicates_to_complete > 0:
+                # Count predicates that are now satisfied but weren't initially
+                newly_satisfied_count = 0
+                for j, predicate in enumerate(option):
+                    current_state = predicate.evaluate()
+                    initial_state = initial_states[j]
+                    # Only count if it went from False to True
+                    if current_state and not initial_state:
+                        newly_satisfied_count += 1
+
+                # Calculate score as proportion of initially unsatisfied predicates that are now satisfied
+                option_score = newly_satisfied_count / predicates_to_complete
+            else:
+                # All predicates were already satisfied initially, no progress to measure
+                option_score = 0.0
```

Comment:
This is unlikely/impossible. No point in having this performance hack. We can remove the predicates_to_complete variable/condition.